Hardcoded credentials panic

Running on a large repo, getting this panic from the hardcoded credentials check.

This is from revision 93016846d28840b0d101431716f317f0a0ee0a88.

```
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x2b9ba0, 0xc420012240)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/GoASTScanner/gas/rules.(*Credentials).matchGenDecl(0xc4201827e0, 0xc4206a7580, 0xc42008e1c0, 0xa40000c4206f2f80, 0x4, 0xc42008cb00)
	/Users/cs/src/go/src/github.com/GoASTScanner/gas/rules/hardcoded_credentials.go:62 +0x23e
github.com/GoASTScanner/gas/rules.(*Credentials).Match(0xc4201827e0, 0x459ea0, 0xc4206a7580, 0xc42008e1c0, 0x307101, 0x5, 0xffffffffffffffff)
	/Users/cs/src/go/src/github.com/GoASTScanner/gas/rules/hardcoded_credentials.go:34 +0x118
github.com/GoASTScanner/gas/core.(*Analyzer).Visit(0xc42008e1b0, 0x459ea0, 0xc4206a7580, 0x4587e0, 0xc42008e1b0)
	/Users/cs/src/go/src/github.com/GoASTScanner/gas/core/analyzer.go:214 +0x277
go/ast.Walk(0x4587e0, 0xc42008e1b0, 0x459ea0, 0xc4206a7580)
	/usr/local/go/src/go/ast/walk.go:52 +0x63
go/ast.Walk(0x4587e0, 0xc42008e1b0, 0x5712e0, 0xc4207ce160)
	/usr/local/go/src/go/ast/walk.go:186 +0x32bf
go/ast.walkStmtList(0x4587e0, 0xc42008e1b0, 0xc42079f200, 0x14, 0x20)
	/usr/local/go/src/go/ast/walk.go:32 +0x89
go/ast.Walk(0x4587e0, 0xc42008e1b0, 0x459c20, 0xc420574c30)
	/usr/local/go/src/go/ast/walk.go:224 +0x1d48
go/ast.Walk(0x4587e0, 0xc42008e1b0, 0x56dcc0, 0xc420574c60)
	/usr/local/go/src/go/ast/walk.go:344 +0xf03
go/ast.walkDeclList(0x4587e0, 0xc42008e1b0, 0xc4201f3080, 0x2, 0x2)
	/usr/local/go/src/go/ast/walk.go:38 +0x89
go/ast.Walk(0x4587e0, 0xc42008e1b0, 0x459e20, 0xc420322d00)
	/usr/local/go/src/go/ast/walk.go:353 +0x29e4
github.com/GoASTScanner/gas/core.(*Analyzer).process(0xc42008e1b0, 0xc42086a550, 0x42, 0x0, 0x0, 0x2800000000000001, 0x8)
	/Users/cs/src/go/src/github.com/GoASTScanner/gas/core/analyzer.go:137 +0x5b7
github.com/GoASTScanner/gas/core.(*Analyzer).Process(0xc42008e1b0, 0xc42086a550, 0x42, 0x3, 0x495a01)
	/Users/cs/src/go/src/github.com/GoASTScanner/gas/core/analyzer.go:159 +0x55
main.main.func1(0xc42086a550, 0x42, 0x45d5a0, 0xc4202865b0, 0x0, 0x0, 0x0, 0x0)
	/Users/cs/src/go/src/github.com/GoASTScanner/gas/main.go:198 +0x15e
path/filepath.walk(0xc42086a550, 0x42, 0x45d5a0, 0xc4202865b0, 0xc42019c000, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:351 +0x81
path/filepath.walk(0xc420412a50, 0x2e, 0x45d5a0, 0xc4202864e0, 0xc42019c000, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:376 +0x344
path/filepath.walk(0xc4203f08d0, 0x29, 0x45d5a0, 0xc4206c05b0, 0xc42019c000, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:376 +0x344
path/filepath.walk(0xc420014004, 0x1e, 0x45d5a0, 0xc42017d5f0, 0xc42019c000, 0x0, 0x1)
	/usr/local/go/src/path/filepath/path.go:376 +0x344
path/filepath.Walk(0xc420014004, 0x1e, 0xc42019c000, 0x0, 0xc420015e01)
	/usr/local/go/src/path/filepath/path.go:398 +0xd5
main.main()
	/Users/cs/src/go/src/github.com/GoASTScanner/gas/main.go:204 +0x12a5
```

This is on a big repo, so I don't know exactly what's killing it yet, but I'll try to narrow it down and see if I can come up with a minimal example.

--- a/rules/hardcoded_credentials.go
+++ b/rules/hardcoded_credentials.go
@@ -58,7 +58,7 @@ func (r *Credentials) matchGenDecl(decl *ast.GenDecl, ctx *gas.Context) (*gas.Is
 	for _, spec := range decl.Specs {
 		if valueSpec, ok := spec.(*ast.ValueSpec); ok {
 			for index, ident := range valueSpec.Names {
-				if r.pattern.MatchString(ident.Name) {
+				if r.pattern.MatchString(ident.Name) && valueSpec.Values != nil {
 					// const foo, bar = "same value"
 					if len(valueSpec.Values) <= index {
 						index = len(valueSpec.Values) - 1

--- a/rules/hardcoded_credentials.go
+++ b/rules/hardcoded_credentials.go
@@ -59,6 +59,10 @@ func (r *Credentials) matchGenDecl(decl *ast.GenDecl, ctx *gas.Context) (*gas.Is
 		if valueSpec, ok := spec.(*ast.ValueSpec); ok {
 			for index, ident := range valueSpec.Names {
 				if r.pattern.MatchString(ident.Name) {
+					// const foo, bar = "same value"
+					if len(valueSpec.Values) <= index {
+						index = len(valueSpec.Values) - 1
+					}
 					if _, ok := valueSpec.Values[index].(*ast.BasicLit); ok {
 						return gas.NewIssue(ctx, decl, r.What, r.Severity, r.Confidence), nil
 					}

--- a/rules/hardcoded_credentials_test.go
+++ b/rules/hardcoded_credentials_test.go
@@ -98,3 +98,16 @@ func TestHardcodedConstantMulti(t *testing.T) {
 
 	checkTestResults(t, issues, 1, "Potential hardcoded credentials")
 }
+
+func TestHardecodedVarsNotAssigned(t *testing.T) {
+	config := map[string]interface{}{"ignoreNosec": false}
+	analyzer := gas.NewAnalyzer(config, nil)
+	analyzer.AddRule(NewHardcodedCredentials(config))
+	issues := gasTestRunner(`
+		package main 
+		var password string
+		func init() {
+			password = "this is a secret string"
+		}`, analyzer)
+	checkTestResults(t, issues, 1, "Potential hardcoded credentials")
+}

--- a/rules/hardcoded_credentials_test.go
+++ b/rules/hardcoded_credentials_test.go
@@ -79,3 +79,22 @@ func TestHardcodedConstant(t *testing.T) {
 
 	checkTestResults(t, issues, 1, "Potential hardcoded credentials")
 }
+
+func TestHardcodedConstantMulti(t *testing.T) {
+	config := map[string]interface{}{"ignoreNosec": false}
+	analyzer := gas.NewAnalyzer(config, nil)
+	analyzer.AddRule(NewHardcodedCredentials(config))
+
+	issues := gasTestRunner(`
+		package samples
+
+		import "fmt"
+
+		const username, password = "secret"
+
+		func main() {
+			fmt.Println("Doing something with: ", username, password)
+		}`, analyzer)
+
+	checkTestResults(t, issues, 1, "Potential hardcoded credentials")
+}

